lib/adapter.js

```diff
@@ -3898,10 +3898,11 @@ function Adapter(options) {
         config.states.maxQueue = config.states.maxQueue || 1000;
 
         // Internal object, but some special adapters want to access it anyway.
-        this.states = new States({
+        new States({
```

Comment:
That's a weird way to use classes IMO.
I'd rather do `const states = new States` and later:
```
connected: () => {
    this.states = states;
}
```

Reply:
Can do, but that "const" var will be in fact never used ...

Tell me a better way to be able to choose in Constructor which kind of instance to return :-)

Reply:
Or other option would be a "static" method like "getInstance" ... but then it would need to return it in a callback or such

Reply:
PS: Your solution would not work because the instance may not be the one from the class. I think I rebuild later into "getInstance" idea ... need to see

Reply:
Aaaaahhh now I remember why the idea was that bad with having an "Init" method ... all the old style testing would break then :-(

Reply:
Well, its not unused. It is later read and assigned to `this.states`. Are you saying that `statesInstance` is not necessarily the one created by `new States`? If so, my approach will not work indeed.

Reply:
Or are you afraid that by invoking the method multiple times, the const may have a different value than intended? That should not happen due to closures.

Reply:
Anyways, we can keep it like you wrote it. It just looks a bit weird in my opinion.